common/courses/notifications.ts

```diff
@@ -136,6 +225,10 @@ export async function sendPupilCoursePromotion(subcourse: Prisma.subcourse, coun
                     },
                 },
             ],
+            notificationPreferences: {
```

Comment:
What is the default for users that do not have this preference?

Reply:
This might also be related to the other comment.

> Why is this removed? Taking all pupils potentially notifies thousands of users, we had problems with this in the past

[Here](https://www.notion.so/lern-fair/Retool-Allow-community-manager-to-promote-courses-manually-c897b7b08cee48318271cab9eb802484?pvs=4#b7e963c8e0d24c05a023d68f85b0a123) is more information about it in the product discussions or @FForstbach could also provide you with more details about this.

But in short, we now want to remove the max limit and at the same time, make some other improvements that should bring a balance, like:
- We moved some of the notification categories that were marked as `system` to `newsletter`, so they should be disabled by default for future users unless they check the `newsletter` checkbox during registration or manually enable the suggestions on their preferences. I tried to tackle that [in this PR](https://github.com/corona-school/backend/pull/1084/files)
- We also recently added a 1-click-opt-out link `/notifications/newsletter?unsubscribe` for newsletters. So it's easier for users to quickly opt out of these emails

> What is the default for users that do not have this preference?

If they don't have this preference set, we shouldn't send them anything, only to those who explicitly agreed to receive these emails. 
I just realized we are already doing this [on the deliverNotification function](https://github.com/corona-school/backend/blob/1096095b09253168864d8cb1cecc9a13610eb35e/common/notification/index.ts#L119). We just have to update the [DEFAULT_PREFERENCES](https://github.com/corona-school/backend/blob/7264c1eb7c000d7b925175fc1545005e8c2506f2/common/notification/defaultPreferences.ts#L45) object which I _(quite embarrassingly)_ missed in my previous PR

However, these measures should heavily impact how many users we reach but also ensure that they're actually interested in these emails.

-------

> What is the default for users that do not have this preference?

Side question, why do we have users who do not have this preference set and depend on the `DEFAULT_PREFERENCES` object? I'm thinking if we should migrate them depending on what they have in the `newsletter` property

Reply:
I would prefer to keep the limit (but increase it if necessary). Just as an additional safeguard in case our assumptions are wrong (or we introduce bugs in the future)

Reply:
With the default preferences we can retrospectively change the defaults without having to update the preferences of all users somehow (potentially overwriting their manually set preference). Since we set them during registration depending on the newsletter field, it lost its value a bit.

Reply:
> I would prefer to keep the limit (but increase it if necessary). Just as an additional safeguard in case our assumptions are wrong (or we introduce bugs in the future)

Ok, I brought up the topic again and it was decided that we'll leave the limit behind a feature flag so we can try different things and see what fits us best.

> With the default preferences we can retrospectively change the defaults without having to update the preferences of all users somehow (potentially overwriting their manually set preference). Since we set them during registration depending on the newsletter field, it lost its value a bit.

I see. Then yeah, it would be nice to migrate those users that don't have the `notificationPreferences` to have the default ones _(which I already updated to be opted-out)_